include/sys/dmu.h

```diff
@@ -532,6 +532,26 @@ void dmu_redact(objset_t *os, uint64_t object, uint64_t offset, uint64_t size,
 void dmu_write_policy(objset_t *os, dnode_t *dn, int level, int wp,
     struct zio_prop *zp);
 
+/*
+ * DB_RF_* are to be used for dbuf_read() or in limited other cases.
```

Comment:
To make it easier to ready, would it be better to either have a separate set of `DBUF_READ_` flags with just the ones that can work, or, have `dbuf_read()` (etc) take `DMU_READ_` flags but then mask off the valid ones?

(again, nbd, it just took me a while to see it. Not least, I didn't even see that the `DB_` and `DMU_` prefixes were different until I start writing this comment :grimacing:)

Reply:
Originally those two sets of flags were independent, declared in different files. But there were several cases where dbuf flags were passed to dmu functions.  Allowing more dmu functions to receive dmu flags meant that I either have to pass two sets of flags, or somehow unify them.  I decided the second, which means that to avoid conflicts they should better be defined in one place.  But same time I am not sure I like unifying the naming, since not all dmu flags are applicable to dbufs, and not all dbuf flags are applicable to dmu.

Reply:
Yeah, I get it - it's not really an easy thing to solve in C. In another language, I'd have two separate enum-style types, and some sort of implicit conversion. It can be done with macros, but because it can't really be enforced it's always going to be a little shaky. I'm cool with leaving it there.